tests/dataset_builders/pie/test_aae2.py

```diff
@@ -0,0 +1,612 @@
+    assert span_texts_labels_tuples[7] == (
+        "the significance of competition is that how to become more excellence to gain the victory",
+        "Premise",
+    )
+    assert span_texts_labels_tuples[8] == (
+        "when we consider about the question that how to win the game, we always find that we need the cooperation",
+        "Premise",
+    )
+    assert span_texts_labels_tuples[9] == (
+        "Take Olympic games which is a form of competition for instance, it is hard to imagine how an athlete could "
+        "win the game without the training of his or her coach, and the help of other professional staffs such as "
+        "the people who take care of his diet, and those who are in charge of the medical care",
+        "Premise",
+    )
+    assert span_texts_labels_tuples[10] == (
+        "without the cooperation, there would be no victory of competition",
+        "Claim",
+    )
+
+    # check relations
+    assert len(document.relations) == 6
+    document.relations[0].label == "supports"
+    document.relations[0].head == document.spans[3]
+    document.relations[0].tail == document.spans[2]
+    document.relations[1].label == "supports"
+    document.relations[1].head == document.spans[4]
+    document.relations[1].tail == document.spans[2]
+    document.relations[2].label == "supports"
+    document.relations[2].head == document.spans[5]
+    document.relations[2].tail == document.spans[2]
+    document.relations[3].label == "supports"
+    document.relations[3].head == document.spans[9]
+    document.relations[3].tail == document.spans[10]
+    document.relations[4].label == "supports"
+    document.relations[4].head == document.spans[8]
+    document.relations[4].tail == document.spans[10]
+    document.relations[5].label == "supports"
+    document.relations[5].head == document.spans[7]
+    document.relations[5].tail == document.spans[8]
+
+
+@pytest.fixture(scope="module")
+def dataset_of_text_documents_with_labeled_spans_and_binary_relations(
+    dataset, dataset_variant
+) -> DatasetDict:
+    if dataset_variant == "default":
+        converted_dataset = dataset.to_document_type(
+            TextDocumentWithLabeledSpansAndBinaryRelations
+        )
+    else:
+        raise ValueError(f"Unknown dataset variant: {dataset_variant}")
+    return converted_dataset
+
+
+@pytest.mark.parametrize("method", ["connect_first", "connect_all"])
+def test_convert_aae2_claim_attributions_to_relations_all(document, method):
+    if dataset_variant == "default" or None:
+        converted_doc = convert_aae2_claim_attributions_to_relations(document, method)
+        converted_binary_relations = converted_doc.binary_relations
+        if method == "connect_first":
+            assert len(converted_binary_relations) == 10
+        elif method == "connect_all":
+            assert len(converted_binary_relations) == 12
+        else:
+            raise ValueError(f"Unknown method: {method}")
```

Comment:
This code is never reached. Please have an `else` case that raises an exception everywhere so that sth like this does not happen.

Reply:
In general, I would recommend to not pass in the `document`, but create it from scratch for this test to make the test more transparent and to directly construct it in the way you need it to cover all parts of the method. E.g.

```python
document = BratDocumentWithMergedSpans(text="This is an example claim. This is a major claim.")
claim1 = LabeledSpan(start=0, end=..., label="claim")
document.spans.append(claim1)
# sanity check (works only after claim1 was added to the document)
assert str(claim1) == "This is an example claim."

claim2 = ....

claim1_attribute = BratAttribute(annotation=claim1, label=...)
document.span_attributes.append(claim1_attribute)

relation1 = BinaryRelation(head=claim1, tail=claim, label=...)
document.binary_relations.append(relation1)

...

converted_doc = convert_aae2_claim_attributions_to_relations(document, method=method)

# check the result
```

Reply:
made some changes in 7fc0897, but not yet added `else` case.